ERCS/erc-7765.md

```diff
@@ -0,0 +1,170 @@
+    /// @param _tokenId  the NFT tokenID.
+    /// @param _privilegeId  the ID of the privileges.
+    function isExercised(
+        address _to,
+        uint256 _tokenId,
+        uint256 _privilegeId
+    ) external view returns (bool _exercised);
+
+    /// @notice This function is to list all privilegeIds of a token.
+    /// @param _tokenId  the NFT tokenID.
+    function getPrivilegeIds(
+        uint256 _tokenId
+    ) external view returns (uint256[] memory privilegeIds);
+
+}
+```
+
+The function `exercisePrivilege` performs the exercise action to a specific privilege of a token. If succeeds, it is expected to emit a `PrivilegeExercised` event.
+
+The function `getPrivilegeIds` provides a way to manage the binding relationship between NFTs and privilegeIds.
+
+The **metadata extension** is OPTIONAL for [EIP-7765](./eip-7765.md) smart contracts. This allows your smart contract to be interrogated for its details about the privileges which your NFTs carry.
+
+```
+pragma solidity >=0.7.0 <0.9.0;
+
+/// @title ERC-7765 Privileged Non-Fungible Tokens Tied To Real World Assets, optional metadata extension
+/// @dev See https://eips.ethereum.org/EIPS/eip-7765
+interface IERC7765Metadata /* is IERC7765 */ {
+
+    /// @notice A distinct Uniform Resource Identifier (URI) for a given privilegeId.
+    /// @dev Throws if `_privilegeId` is not a valid privilegeId. URIs are defined in RFC
+    ///  3986. The URI may point to a JSON file that conforms to the "ERC-7765
+    ///  Metadata JSON Schema".
+    function privilegeURI(uint256 _privilegeId) external view returns (string memory);
+
+}
+```
+
+This is the “EIP-7765 Metadata JSON Schema” referenced above.
+
+```
+{
+    "title": "Privilege Metadata",
+    "type": "object",
+    "properties": {
+        "name": {
+            "type": "string",
+            "description": "Identifies the specific privilege."
+        },
+        "description": {
+            "type": "string",
+            "description": "Describes the specific privilege."
+        },
+        "resource": {
+            "type": "string",
+            "description": "A URI pointing to a resource representing the specific privilege."
+        }
+    }
+}
+```
+
+`IERC7765Metadata` provides specifications for obtaining metadata information of privileges. A contract that implements `IERC7765Metadata` **SHALL** also implement `IERC7765`.
+
+## Rationale
+
+1.  With the `PrivilegeExercised` event emitted onchain, we can determine that the user has confirmed the exercise of this privilege, so as to implement the privilege in the real world.
+
+2. We choose to include an address `_to` for functions `exercisePrivilege`, `isExercisable` and `isExercised` so that a specific privilege of an NFT MAY be exercised for someone who will benefit from it other than the NFT holder nor the transaction initiator. And This EIP doesn't assume who has the power to perform this action, it's totally decided by the developers who are using this standard.
```

Comment:
> [...] MAY [...]

The all-caps keywords create requirements, and so should be in the specification section. It seems like you don't intend to create a requirement here, so I'd just change it to the lowercase "may":

```suggestion
2. We choose to include an address `_to` for functions `exercisePrivilege`, `isExercisable` and `isExercised` so that a specific privilege of an NFT may be exercised for someone who will benefit from it other than the NFT holder nor the transaction initiator. This EIP doesn't assume who has the power to perform this action, it's totally decided by the developers who are using this standard.
```